src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
 // write code here
+// Helper function to convert salary to number
+function getSalary(element) {
+  return Number(element.dataset.salary);
+}
+
+// Function to sort list by salary in descending order
+// eslint-disable-next-line no-shadow
+function sortList(list) {
+  const items = Array.from(list.children);
+
+  items.sort((a, b) => getSalary(b) - getSalary(a));
+  items.forEach((item) => list.appendChild(item));
```

Comment:
Appending items back to the list using `list.appendChild(item)` in a loop can result in unexpected behavior in some browsers, as it may not reorder the elements as intended. Consider using `list.innerHTML = '';` before the loop to clear the list, or use a document fragment to append all items at once.